src/Compiler/Checking/CheckComputationExpressions.fs

```diff
@@ -2210,14 +2210,15 @@ let TcSequenceExpression (cenv: cenv) env tpenv comp (overallTy: OverallTy) m =
 
             let env = { env with eContextInfo = ContextInfo.SequenceExpression genOuterTy }
 
-            if enableImplicitYield then 
+            if enableImplicitYield then
                 let hasTypeUnit, expr, tpenv = TryTcStmt cenv env tpenv comp
                 if hasTypeUnit then 
                     Choice2Of2 expr, tpenv
                 else
                     let genResultTy = NewInferenceType g
                     let mExpr = expr.Range
                     UnifyTypes cenv env mExpr genOuterTy (mkSeqTy cenv.g genResultTy)
+                    let expr, tpenv = TcExprFlex cenv flex true genResultTy env tpenv comp
```

Comment:
A bit worried that this is legacy way of doing this, according to the comment here https://github.com/dotnet/fsharp/blob/739acb3a598a5d22c26132e1547afe842b7cfd83/src/Compiler/Checking/CheckExpressions.fs#L5126
But so far idk what is the current way.

Reply:
We are essentially checking the same SynExpr (`comp`) twice, second time with a subtype constraint (first time to check if its type is `unit`)